{"cells":[{"cell_type":"markdown","metadata":{"id":"6qMNTZCfXa8I"},"source":["1ï¸âƒ£ Install IQ-TREE\n"]},{"cell_type":"code","execution_count":12,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":2147,"status":"ok","timestamp":1743260649301,"user":{"displayName":"Mert MermeroÄŸlu","userId":"09814328276549859854"},"user_tz":-60},"id":"CG_i5_-XYcB_","outputId":"ca4e1ef6-b25a-4137-b9f1-fb63dc66e7d1"},"outputs":[{"output_type":"stream","name":"stdout","text":["Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount(\"/content/drive\", force_remount=True).\n"]}],"source":["from google.colab import drive\n","drive.mount('/content/drive')"]},{"cell_type":"markdown","metadata":{"id":"FuXAjm3YeKJ_"},"source":["#ðŸ§¬ Align Sequences (if needed)\n","If your sequences are not aligned, install and use MAFFT to align them:"]},{"cell_type":"code","execution_count":13,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":49997,"status":"ok","timestamp":1743260702327,"user":{"displayName":"Mert MermeroÄŸlu","userId":"09814328276549859854"},"user_tz":-60},"id":"wIlG-qS-Y5-Q","outputId":"506ec173-262e-4d86-a8b2-70d4dbdb5144"},"outputs":[{"output_type":"stream","name":"stdout","text":["Reading package lists... Done\n","Building dependency tree... Done\n","Reading state information... Done\n","iqtree is already the newest version (2.0.7+dfsg-1).\n","mafft is already the newest version (7.490-1).\n","0 upgraded, 0 newly installed, 0 to remove and 29 not upgraded.\n","nthread = 0\n","nthreadpair = 0\n","nthreadtb = 0\n","ppenalty_ex = 0\n","stacksize: 8192 kb\n","rescale = 1\n","Gap Penalty = -1.53, +0.00, +0.00\n","\n","\n","\n","Making a distance matrix ..\n","\n","There are 2 ambiguous characters.\n","    1 / 17\n","done.\n","\n","Constructing a UPGMA tree (efffree=0) ... \n","   10 / 17\n","done.\n","\n","Progressive alignment 1/2... \n","STEP    14 / 16  f\b\b\n","Reallocating..done. *alloclen = 13497\n","STEP    16 / 16  f\b\b\n","done.\n","\n","Making a distance matrix from msa.. \n","    0 / 17\n","done.\n","\n","Constructing a UPGMA tree (efffree=1) ... \n","   10 / 17\n","done.\n","\n","Progressive alignment 2/2... \n","STEP     6 / 16  f\b\b\n","Reallocating..done. *alloclen = 13287\n","STEP    16 / 16  f\b\b\n","done.\n","\n","disttbfast (aa) Version 7.490\n","alg=A, model=BLOSUM62, 1.53, -0.00, -0.00, noshift, amax=0.0\n","0 thread(s)\n","\n","rescale = 1\n","dndpre (aa) Version 7.490\n","alg=X, model=BLOSUM62, 1.53, +0.12, -0.00, noshift, amax=0.0\n","0 thread(s)\n","\n","minimumweight = 0.000010\n","autosubalignment = 0.000000\n","nthread = 0\n","randomseed = 0\n","blosum 62 / kimura 200\n","poffset = 0\n","niter = 2\n","sueff_global = 0.100000\n","nadd = 2\n","rescale = 1\n","\n","   10 / 17\n","Segment   1/  1    1-7401\n","done\n","dvtditr (aa) Version 7.490\n","alg=A, model=BLOSUM62, 1.53, -0.00, -0.00, noshift, amax=0.0\n","0 thread(s)\n","\n","\n","Strategy:\n"," FFT-NS-i (Standard)\n"," Iterative refinement method (max. 2 iterations)\n","\n","If unsure which option to use, try 'mafft --auto input > output'.\n","For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n","\n","The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n","It tends to insert more gaps into gap-rich regions than previous versions.\n","To disable this change, add the --leavegappyregion option.\n","\n"]}],"source":["!apt-get install mafft iqtree\n","!mafft --auto /content/drive/MyDrive/gobi/deneme/umapFasta.fna > /content/drive/MyDrive/gobi/deneme/denemeMAFFT.fasta"]},{"cell_type":"markdown","metadata":{"id":"yAbT0Xdi69aC"},"source":["#PHY"]},{"cell_type":"code","execution_count":14,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":3127,"status":"ok","timestamp":1743260709396,"user":{"displayName":"Mert MermeroÄŸlu","userId":"09814328276549859854"},"user_tz":-60},"id":"LwQzmCfc7HtL","outputId":"b2dc4337-cffc-44e2-d6bd-c12542aaf12e"},"outputs":[{"output_type":"stream","name":"stdout","text":["Requirement already satisfied: Bio in /usr/local/lib/python3.11/dist-packages (1.7.1)\n","Requirement already satisfied: biopython>=1.80 in /usr/local/lib/python3.11/dist-packages (from Bio) (1.85)\n","Requirement already satisfied: gprofiler-official in /usr/local/lib/python3.11/dist-packages (from Bio) (1.0.0)\n","Requirement already satisfied: mygene in /usr/local/lib/python3.11/dist-packages (from Bio) (3.2.2)\n","Requirement already satisfied: pandas in /usr/local/lib/python3.11/dist-packages (from Bio) (2.2.2)\n","Requirement already satisfied: pooch in /usr/local/lib/python3.11/dist-packages (from Bio) (1.8.2)\n","Requirement already satisfied: requests in /usr/local/lib/python3.11/dist-packages (from Bio) (2.32.3)\n","Requirement already satisfied: tqdm in /usr/local/lib/python3.11/dist-packages (from Bio) (4.67.1)\n","Requirement already satisfied: numpy in /usr/local/lib/python3.11/dist-packages (from biopython>=1.80->Bio) (2.0.2)\n","Requirement already satisfied: biothings-client>=0.2.6 in /usr/local/lib/python3.11/dist-packages (from mygene->Bio) (0.4.1)\n","Requirement already satisfied: python-dateutil>=2.8.2 in /usr/local/lib/python3.11/dist-packages (from pandas->Bio) (2.8.2)\n","Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.11/dist-packages (from pandas->Bio) (2025.1)\n","Requirement already satisfied: tzdata>=2022.7 in /usr/local/lib/python3.11/dist-packages (from pandas->Bio) (2025.1)\n","Requirement already satisfied: platformdirs>=2.5.0 in /usr/local/lib/python3.11/dist-packages (from pooch->Bio) (4.3.7)\n","Requirement already satisfied: packaging>=20.0 in /usr/local/lib/python3.11/dist-packages (from pooch->Bio) (24.2)\n","Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.11/dist-packages (from requests->Bio) (3.4.1)\n","Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.11/dist-packages (from requests->Bio) (3.10)\n","Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.11/dist-packages (from requests->Bio) (2.3.0)\n","Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.11/dist-packages (from requests->Bio) (2025.1.31)\n","Requirement already satisfied: httpx>=0.22.0 in /usr/local/lib/python3.11/dist-packages (from biothings-client>=0.2.6->mygene->Bio) (0.28.1)\n","Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.11/dist-packages (from python-dateutil>=2.8.2->pandas->Bio) (1.17.0)\n","Requirement already satisfied: anyio in /usr/local/lib/python3.11/dist-packages (from httpx>=0.22.0->biothings-client>=0.2.6->mygene->Bio) (4.9.0)\n","Requirement already satisfied: httpcore==1.* in /usr/local/lib/python3.11/dist-packages (from httpx>=0.22.0->biothings-client>=0.2.6->mygene->Bio) (1.0.7)\n","Requirement already satisfied: h11<0.15,>=0.13 in /usr/local/lib/python3.11/dist-packages (from httpcore==1.*->httpx>=0.22.0->biothings-client>=0.2.6->mygene->Bio) (0.14.0)\n","Requirement already satisfied: sniffio>=1.1 in /usr/local/lib/python3.11/dist-packages (from anyio->httpx>=0.22.0->biothings-client>=0.2.6->mygene->Bio) (1.3.1)\n","Requirement already satisfied: typing_extensions>=4.5 in /usr/local/lib/python3.11/dist-packages (from anyio->httpx>=0.22.0->biothings-client>=0.2.6->mygene->Bio) (4.12.2)\n"]}],"source":["!pip install Bio"]},{"cell_type":"code","execution_count":15,"metadata":{"executionInfo":{"elapsed":106,"status":"ok","timestamp":1743260713021,"user":{"displayName":"Mert MermeroÄŸlu","userId":"09814328276549859854"},"user_tz":-60},"id":"EUIS_K4h6_Nb"},"outputs":[],"source":["from Bio import SeqIO\n","import os\n","\n","def fasta_to_phy(fasta_file, phy_file):\n","    \"\"\"\n","    Converts a FASTA file to a Phylip file using Biopython's SeqIO.\n","\n","    Parameters:\n","    fasta_file (str): Path to the input FASTA file.\n","    phy_file (str): Path to the output Phylip file.\n","    \"\"\"\n","\n","    # Read the FASTA file\n","    records = list(SeqIO.parse(fasta_file, \"fasta\"))\n","\n","    # Write to a Phylip file\n","    SeqIO.write(records, phy_file, \"phylip\")\n","\n","# Example usage\n","fasta_file = \"/content/drive/MyDrive/gobi/deneme/denemeMAFFT.fasta\" # Replace with your FASTA file path\n","phy_file = \"/content/drive/MyDrive/gobi/deneme/denemeMAFFT.phy\" # Replace with your desired output file path\n","fasta_to_phy(fasta_file, phy_file)"]},{"cell_type":"markdown","metadata":{"id":"JA7C54Bm9zsM"},"source":["##kullanmadÄ±k (keyfi)"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"1F_ye-sM7d03"},"outputs":[],"source":["def split_fasta(input_fasta, output_folder):\n","    \"\"\"\n","    Split a FASTA file into multiple FASTA files, each containing one sequence, and save them to a specified folder.\n","\n","    Parameters:\n","    - input_fasta: Path to the input FASTA file.\n","    - output_folder: Path to the folder where output FASTA files will be saved.\n","    \"\"\"\n","\n","    # Ensure the output folder exists, create it if not\n","    if not os.path.exists(output_folder):\n","        os.makedirs(output_folder)\n","\n","    # Open the input FASTA file\n","    with open(input_fasta, 'r') as file:\n","        content = file.read()\n","\n","    # Split the file content by '>' to separate sequences, skipping the first empty element if file starts with '>'\n","    sequences = content.split('>')[1:]\n","\n","    for seq in sequences:\n","        # Split each sequence into header and sequence parts\n","        parts = seq.split('\\n', 1)\n","        header = parts[0].strip().replace(' ', '_').replace('|', '_')\n","        sequence = parts[1].strip()\n","\n","        # Create a file name based on the header and ensure it is saved in the output folder\n","        filename = os.path.join(output_folder, f\"{header}.fasta\")\n","\n","        # Write the sequence to its own file\n","        with open(filename, 'w') as output_file:\n","            output_file.write(f\">{header}\\n{sequence}\")\n","        input_fasta = \"/content/drive/MyDrive/gobi/Allaligned.fasta\" # Replace with your FASTA file path\n","        output_file = \"/content/drive/MyDrive/gobi/selin/output\"\n","        split_fasta(input_fasta_path, output_folder_path)"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"59HnhRcM7eT3"},"outputs":[],"source":["input_fasta_path = '/Users/julius/Gobi/fetch_data/Output/sequences_Timeout_and_timeless_Final.fasta'\n","output_folder_path = '/Users/julius/Gobi/fetch_data/single_fasta_sequences'\n","\n","split_fasta(input_fasta_path, output_folder_path)"]},{"cell_type":"code","execution_count":20,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"kUKxgKcu9zLB","outputId":"3ced1c9e-bc84-4212-9d0d-81fb1a8c3cd3","executionInfo":{"status":"ok","timestamp":1743261966476,"user_tz":-60,"elapsed":1052555,"user":{"displayName":"Mert MermeroÄŸlu","userId":"09814328276549859854"}}},"outputs":[{"output_type":"stream","name":"stdout","text":["IQ-TREE multicore version 2.0.7 for Linux 64-bit built Jan 21 2022\n","Developed by Bui Quang Minh, Nguyen Lam Tung, Olga Chernomor,\n","Heiko Schmidt, Dominik Schrempf, Michael Woodhams.\n","\n","Host:    95c9b48ea070 (AVX2, FMA3, 12 GB RAM)\n","Command: iqtree2 -s /content/drive/MyDrive/gobi/deneme/denemeMAFFT.phy -m LG+F+G -bb 1000 -nt AUTO\n","Seed:    962325 (Using SPRNG - Scalable Parallel Random Number Generator)\n","Time:    Sat Mar 29 15:08:33 2025\n","Kernel:  AVX+FMA - auto-detect threads (2 CPU cores detected)\n","\n","Reading alignment file /content/drive/MyDrive/gobi/deneme/denemeMAFFT.phy ... Phylip format detected\n","Alignment most likely contains protein sequences\n","WARNING: 1 sites contain only gaps or ambiguous characters.\n","Alignment has 17 sequences with 7263 columns, 5147 distinct patterns\n","2020 parsimony-informative, 3377 singleton sites, 1866 constant sites\n","            Gap/Ambiguity  Composition  p-value\n","   1  2_Bombyx_m   27.54%    failed      0.00%\n","   2  Cydia        89.16%    failed      0.00%\n","   3  2_Manduca_   62.89%    failed      0.00%\n","   4  2_Plutella   36.82%    failed      0.00%\n","   5  3_Homalodi   84.28%    failed      0.00%\n","   6  Achroia_gr   32.38%    failed      0.00%\n","   7  Apis_melli   79.58%    failed      0.00%\n","   8  Bombus_imp   94.86%    failed      0.00%\n","   9  Bombyx       93.83%    failed      0.00%\n","  10  Cotesia_gl   88.93%    failed      0.00%\n","  11  Galleria_m   17.42%    failed      0.00%\n","  12  Halyomorph   90.82%    failed      0.00%\n","  13  Maniola_ju   67.92%    failed      0.00%\n","  14  Monomorium   86.52%    failed      0.00%\n","  15  Nilaparvat   88.30%    failed      0.00%\n","  16  Vanessa_ca   40.44%    failed      0.00%\n","  17  Vespula_vu   96.37%    failed      0.00%\n","WARNING: 12 sequences contain more than 50% gaps/ambiguity\n","****  TOTAL        69.30%  17 sequences failed composition chi2 test (p-value<5%; df=19)\n","\n","Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.025 seconds\n","Generating 1000 samples for ultrafast bootstrap (seed: 962325)...\n","\n","NOTE: 74 MB RAM (0 GB) is required!\n","Measuring multi-threading efficiency up to 2 CPU cores\n","4 trees examined\n","Threads: 1 / Time: 2.074 sec / Speedup: 1.000 / Efficiency: 100% / LogL: -76424\n","Threads: 2 / Time: 3.456 sec / Speedup: 0.600 / Efficiency: 30% / LogL: -76424\n","BEST NUMBER OF THREADS: 1\n","\n","Estimate model parameters (epsilon = 0.100)\n","1. Initial log-likelihood: -83333.814\n","2. Current log-likelihood: -74066.571\n","3. Current log-likelihood: -74004.011\n","4. Current log-likelihood: -74001.940\n","5. Current log-likelihood: -74001.273\n","6. Current log-likelihood: -74001.054\n","Optimal log-likelihood: -74000.944\n","Gamma shape alpha: 1.019\n","Parameters optimization took 6 rounds (10.968 sec)\n","Computing ML distances based on estimated model parameters... 0.141 sec\n","WARNING: Some pairwise ML distances are too long (saturated)\n","Computing BIONJ tree...\n","0.017 seconds\n","Log-likelihood of BIONJ tree: -74054.298\n","--------------------------------------------------------------------\n","|             INITIALIZING CANDIDATE TREE SET                      |\n","--------------------------------------------------------------------\n","Generating 98 parsimony trees... 2.510 second\n","Computing log-likelihood of 98 initial trees ... 44.445 seconds\n","Current best score: -74000.944\n","\n","Do NNI search on 20 best initial trees\n","Estimate model parameters (epsilon = 0.100)\n","BETTER TREE FOUND at iteration 1: -73985.396\n","Estimate model parameters (epsilon = 0.100)\n","BETTER TREE FOUND at iteration 3: -73961.554\n","Estimate model parameters (epsilon = 0.100)\n","BETTER TREE FOUND at iteration 10: -73959.460\n","Iteration 10 / LogL: -73959.460 / Time: 0h:2m:51s\n","Estimate model parameters (epsilon = 0.100)\n","BETTER TREE FOUND at iteration 11: -73959.257\n","Iteration 20 / LogL: -73959.560 / Time: 0h:4m:40s\n","Finish initializing candidate tree set (7)\n","Current best tree score: -73959.257 / CPU time: 255.585\n","Number of iterations: 20\n","--------------------------------------------------------------------\n","|               OPTIMIZING CANDIDATE TREE SET                      |\n","--------------------------------------------------------------------\n","Iteration 30 / LogL: -73960.037 / Time: 0h:6m:4s (0h:16m:58s left)\n","Iteration 40 / LogL: -73959.763 / Time: 0h:7m:17s (0h:13m:16s left)\n","Iteration 50 / LogL: -73959.327 / Time: 0h:8m:43s (0h:10m:51s left)\n","Log-likelihood cutoff on original alignment: -74012.901\n","Iteration 60 / LogL: -73959.498 / Time: 0h:10m:0s (0h:8m:39s left)\n","Iteration 70 / LogL: -73959.419 / Time: 0h:11m:23s (0h:6m:46s left)\n","Iteration 80 / LogL: -73959.662 / Time: 0h:12m:47s (0h:5m:1s left)\n","Iteration 90 / LogL: -73959.739 / Time: 0h:14m:18s (0h:3m:22s left)\n","Iteration 100 / LogL: -73961.379 / Time: 0h:15m:43s (0h:1m:44s left)\n","Log-likelihood cutoff on original alignment: -74012.901\n","NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.997\n","Iteration 110 / LogL: -73959.356 / Time: 0h:17m:4s (0h:14m:6s left)\n","TREE SEARCH COMPLETED AFTER 112 ITERATIONS / Time: 0h:17m:24s\n","\n","--------------------------------------------------------------------\n","|                    FINALIZING TREE SEARCH                        |\n","--------------------------------------------------------------------\n","Performs final model parameters optimization\n","Estimate model parameters (epsilon = 0.010)\n","1. Initial log-likelihood: -73959.257\n","Optimal log-likelihood: -73959.254\n","Gamma shape alpha: 1.040\n","Parameters optimization took 1 rounds (1.152 sec)\n","BEST SCORE FOUND : -73959.254\n","Creating bootstrap support values...\n","Split supports printed to NEXUS file /content/drive/MyDrive/gobi/deneme/denemeMAFFT.phy.splits.nex\n","Total tree length: 37.692\n","\n","Total number of iterations: 112\n","CPU time used for tree search: 1005.310 sec (0h:16m:45s)\n","Wall-clock time used for tree search: 1018.999 sec (0h:16m:58s)\n","Total CPU time used: 1031.545 sec (0h:17m:11s)\n","Total wall-clock time used: 1045.406 sec (0h:17m:25s)\n","\n","Computing bootstrap consensus tree...\n","Reading input file /content/drive/MyDrive/gobi/deneme/denemeMAFFT.phy.splits.nex...\n","17 taxa and 124 splits.\n","Consensus tree written to /content/drive/MyDrive/gobi/deneme/denemeMAFFT.phy.contree\n","Reading input trees file /content/drive/MyDrive/gobi/deneme/denemeMAFFT.phy.contree\n","Log-likelihood of consensus tree: -73959.363\n","\n","Analysis results written to: \n","  IQ-TREE report:                /content/drive/MyDrive/gobi/deneme/denemeMAFFT.phy.iqtree\n","  Maximum-likelihood tree:       /content/drive/MyDrive/gobi/deneme/denemeMAFFT.phy.treefile\n","  Likelihood distances:          /content/drive/MyDrive/gobi/deneme/denemeMAFFT.phy.mldist\n","\n","Ultrafast bootstrap approximation results written to:\n","  Split support values:          /content/drive/MyDrive/gobi/deneme/denemeMAFFT.phy.splits.nex\n","  Consensus tree:                /content/drive/MyDrive/gobi/deneme/denemeMAFFT.phy.contree\n","  Screen log file:               /content/drive/MyDrive/gobi/deneme/denemeMAFFT.phy.log\n","\n","Date and Time: Sat Mar 29 15:26:06 2025\n"]}],"source":["!iqtree2 -s /content/drive/MyDrive/gobi/deneme/denemeMAFFT.phy -m LG+F+G -bb 1000 -nt AUTO"]},{"cell_type":"code","execution_count":21,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":112,"status":"ok","timestamp":1743262086043,"user":{"displayName":"Mert MermeroÄŸlu","userId":"09814328276549859854"},"user_tz":-60},"id":"naV4UoUkLNdi","outputId":"5b9a6b56-0916-4bf8-b887-612bbf20176e"},"outputs":[{"output_type":"stream","name":"stdout","text":["IQ-TREE multicore version 2.0.7 for Linux 64-bit built Jan 21 2022\n","Developed by Bui Quang Minh, Nguyen Lam Tung, Olga Chernomor,\n","Heiko Schmidt, Dominik Schrempf, Michael Woodhams.\n","\n","Command-line examples (replace 'iqtree2 ...' by actual path to executable):\n","\n","1. Infer maximum-likelihood tree from a sequence alignment (example.phy)\n","   with the best-fit model automatically selected by ModelFinder:\n","     iqtree2 -s example.phy\n","\n","2. Perform ModelFinder without subsequent tree inference:\n","     iqtree2 -s example.phy -m MF\n","   (use '-m TEST' to resemble jModelTest/ProtTest)\n","\n","3. Combine ModelFinder, tree search, ultrafast bootstrap and SH-aLRT test:\n","     iqtree2 -s example.phy --alrt 1000 -B 1000\n","\n","4. Perform edge-linked proportional partition model (example.nex):\n","     iqtree2 -s example.phy -p example.nex\n","   (replace '-p' by '-Q' for edge-unlinked model)\n","\n","5. Find best partition scheme by possibly merging partitions:\n","     iqtree2 -s example.phy -p example.nex -m MF+MERGE\n","   (use '-m TESTMERGEONLY' to resemble PartitionFinder)\n","\n","6. Find best partition scheme followed by tree inference and bootstrap:\n","     iqtree2 -s example.phy -p example.nex -m MFP+MERGE -B 1000\n","\n","7. Use 4 CPU cores to speed up computation: add '-T 4' option\n","\n","8. Polymorphism-aware model with HKY nucleotide model and Gamma rate:\n","     iqtree2 -s counts_file.cf -m HKY+P+G\n","\n","9. PoMo mixture with virtual popsize 5 and weighted binomial sampling:\n","     iqtree2 -s counts_file.cf -m \"MIX{HKY+P{EMP},JC+P}+N5+WB\"\n","\n","To show all available options: run 'iqtree2 -h'\n","\n","Have a look at the tutorial and manual for more information:\n","     http://www.iqtree.org\n","\n"]}],"source":["!iqtree2"]},{"cell_type":"code","execution_count":22,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":3406,"status":"ok","timestamp":1743262092601,"user":{"displayName":"Mert MermeroÄŸlu","userId":"09814328276549859854"},"user_tz":-60},"id":"lkvMK1sOMOR9","outputId":"dddb53a8-f967-461d-c3b3-7affd8f9af40"},"outputs":[{"output_type":"stream","name":"stdout","text":["Requirement already satisfied: ete3 in /usr/local/lib/python3.11/dist-packages (3.1.3)\n"]}],"source":["!pip install --upgrade ete3"]},{"cell_type":"code","execution_count":26,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":14,"status":"ok","timestamp":1743262667771,"user":{"displayName":"Mert MermeroÄŸlu","userId":"09814328276549859854"},"user_tz":-60},"id":"jrWJq0syMMSS","outputId":"0a7e62f1-b3ee-4a5b-c171-7dc7923652f7"},"outputs":[{"output_type":"stream","name":"stdout","text":["\n","   /-2_Bombyx_m\n","  |\n","  |                           /-Cydia\n","  |                        /-|\n","  |                       |   \\-Monomorium\n","  |                       |\n","  |                     /-|      /-3_Homalodi\n","  |                    |  |   /-|\n","  |                    |  |  |  |   /-Bombus_imp\n","  |                    |   \\-|   \\-|\n","  |                  /-|     |      \\-Halyomorph\n","  |                 |  |     |\n","  |                 |  |      \\-Apis_melli\n","  |                 |  |\n","  |               /-|  |   /-Cotesia_gl\n","  |              |  |   \\-|\n","--|              |  |      \\-Vespula_vu\n","  |            /-|  |\n","  |           |  |   \\-Maniola_ju\n","  |           |  |\n","  |         /-|   \\-2_Manduca_\n","  |        |  |\n","  |        |  |   /-Achroia_gr\n","  |      /-|   \\-|\n","  |     |  |      \\-Galleria_m\n","  |   /-|  |\n","  |  |  |   \\-2_Plutella\n","  |--|  |\n","  |  |   \\-Vanessa_ca\n","  |  |\n","  |   \\-Nilaparvat\n","  |\n","   \\-Bombyx\n"]}],"source":["from ete3 import Tree\n","\n","# Load the tree from IQ-TREE's output file\n","input_file = '/content/drive/MyDrive/gobi/deneme/denemeMAFFT.phy.contree'\n","\n","tree = Tree(input_file)\n","\n","\n","# Render the tree inline in Colab\n","print(tree)"]},{"cell_type":"code","execution_count":null,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":20469,"status":"ok","timestamp":1742145709063,"user":{"displayName":"Selin","userId":"07842812702483837140"},"user_tz":-60},"id":"KUmd_io18dpF","outputId":"ec758b9e-b33b-4ab7-870c-28c801c11fcb"},"outputs":[{"data":{"text/plain":["189"]},"execution_count":5,"metadata":{},"output_type":"execute_result"}],"source":["from Bio import AlignIO, SeqIO\n","from Bio.Align import MultipleSeqAlignment\n","from Bio.SeqRecord import SeqRecord\n","from Bio.Seq import Seq\n","\n","\n","input_file = \"/content/drive/MyDrive/gobi/Allaligned.fasta\"\n","\n","alignment = AlignIO.read(input_file, \"fasta\")\n","\n","alignment_length = len(alignment[0].seq)\n","num_seqs = len(alignment)\n","treshold = 0.3\n","\n","columns_to_be_deleted = []\n","\n","# Identify columns for deletion based on first sequence's characters\n","for column_index in range(alignment_length):\n","    ref_char = alignment[0].seq[column_index]\n","    match_count = sum(1 for seq in alignment if seq.seq[column_index] == ref_char)\n","\n","    # Delete columns if the character from the first sequence is in less than 50% sequences\n","    if match_count / num_seqs < treshold:\n","        columns_to_be_deleted.append(column_index)\n","\n","# Build new sequences excluding the columns marked for deletion\n","new_records = []\n","for seq_record in alignment:\n","    new_seq = ''.join(\n","        char for idx, char in enumerate(seq_record.seq)\n","        if idx not in columns_to_be_deleted\n","    )\n","    new_records.append(SeqRecord(Seq(new_seq), id=seq_record.id, description=\"\"))\n","\n","# Save the new alignment to a FASTA file\n","SeqIO.write(new_records, \"filtered_alignment.fasta\", \"fasta\")"]},{"cell_type":"markdown","source":["Get rid of some unneccassary gaps"],"metadata":{"id":"BZzMK0KEUXpW"}},{"cell_type":"code","execution_count":null,"metadata":{"colab":{"base_uri":"https://localhost:8080/","height":216},"executionInfo":{"elapsed":11,"status":"error","timestamp":1742302731243,"user":{"displayName":"Selin","userId":"07842812702483837140"},"user_tz":-60},"id":"yK2hbRKlCWNt","outputId":"42159109-bf70-448b-a778-ebf086dac5e7"},"outputs":[{"output_type":"error","ename":"NameError","evalue":"name 'alignment' is not defined","traceback":["\u001b[0;31m---------------------------------------------------------------------------\u001b[0m","\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)","\u001b[0;32m<ipython-input-1-6f1e25c12469>\u001b[0m in \u001b[0;36m<cell line: 0>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;31m# Identify columns where all sequences have '-'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0mcolumns_to_remove\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 3\u001b[0;31m \u001b[0mseq_length\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0malignment\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mseq\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      4\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mj\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mrange\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mseq_length\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m  \u001b[0;31m# Iterate through columns\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;31mNameError\u001b[0m: name 'alignment' is not defined"]}],"source":["# Identify columns where all sequences have '-'\n","columns_to_remove = []\n","seq_length = len(alignment[0].seq)\n","\n","for j in range(seq_length):  # Iterate through columns\n","    if all(record.seq[j] == '-' for record in alignment):  # If all are '-'\n","        columns_to_remove.append(j)\n","\n","print(columns_to_remove)\n","# Create new filtered alignment without those columns\n","filtered_alignment = [\n","    SeqRecord(\n","        Seq(\"\".join(record.seq[i] for i in range(seq_length) if i not in columns_to_remove)),\n","        id=record.id,\n","        description=\"\"\n","    )\n","    for record in filtered_alignment\n","]\n","\n","# Return the modified filtered_alignment\n","print(filtered_alignment)\n","\n","\n","# âœ… Use SeqIO.write() to save as FASTA\n","SeqIO.write(filtered_alignment, output_file, \"fasta\")\n","\n","print(f\"Filtered sequences saved as {output_file}\")"]},{"cell_type":"code","source":[],"metadata":{"id":"drDm73kRVdqF"},"execution_count":null,"outputs":[]}],"metadata":{"colab":{"provenance":[]},"kernelspec":{"display_name":"Python 3","name":"python3"},"language_info":{"name":"python"}},"nbformat":4,"nbformat_minor":0}